### PR TITLE
Add temperature sensors to YarpSensorBridge and YarpRobotLogger, migrate to MAS interfaces for FT Sensors

### PR DESCRIPTION
**Pull request by prashanthr05**
cc @isorrentino @GiulioRomualdi

### CODE REVIEW

**Comment by DanielePucci** [comment]
Once ready, this PR is interesting also to @SanderRuijters @S-Dafarra since we may perform calibration experiments with different temperature and calibrate the sensors against it, analogously to what we did in [Model Based In Situ Calibration with Temperature compensation of 6 axis Force Torque Sensors](https://arxiv.org/pdf/1812.00650.pdf)

@prashanthr05 how's the status of this PR?

**Comment by prashanthr05** [comment]
> @prashanthr05 how's the status of this PR?

The code changes are ready. This PR needs to be tested on the robot. 

Currently, I've introduced changes in the configuration files only to log temperatures from the arm FTs of `iCubGenova09`.

**Comment by GiulioRomualdi** [comment]
Not related to this PR but we may think to avoid asking for the temperature every time the `getFeedback()` is called. Indeed:
1. If I'm not mistaken the temperature is retrieved by rpc (so it will slow down the main application. similar issue: https://github.com/ami-iit/bipedal-locomotion-framework/issues/499)
2. the temperature dynamics is pretty slow with respect to the other quantities we use to collect.

**Comment by SanderRuijters** [comment]
> This PR needs to be tested on the robot.

@prashanthr05 could you maybe notify me when you are going to perform these tests? I am curious to know more about this implementation and possible results.

**Comment by prashanthr05** [comment]
> > This PR needs to be tested on the robot.
> 
> @prashanthr05 could you maybe notify me when you are going to perform these tests? I am curious to know more about this implementation and possible results.

Ciao @SanderRuijters !! Sure, I was planning to do it this week, I will notify you when I schedule this activity.

**Comment by isorrentino** [comment]
I pushed the modifications discussed yesterday in https://github.com/ami-iit/bipedal-locomotion-framework/pull/519/commits/af202fb7b9b067f7801eea8b8243f5cea730f1da and https://github.com/ami-iit/bipedal-locomotion-framework/pull/519/commits/71d31e28e5fe8269e217499586f44afc2ef9ac3d.

cc @prashanthr05

File: devices/YarpRobotLoggerDevice/src/YarpRobotLoggerDevice.cpp [hunk 1]
```diff
@@ -403,6 +403,14 @@ void YarpRobotLoggerDevice::run()
...
     }
 
+    for (const auto& sensorname : m_robotSensorBridge->getTemperatureSensorsList())
+    {
+        if (m_robotSensorBridge->getTemperature(sensorname, m_ftTemperatureBuffer))
+        {
+            m_bufferManager.push_back({m_ftTemperatureBuffer}, time, "temperatures::" + sensorname);
+        }
+    }
+
```
Comment (prashanthr05) [inline devices/YarpRobotLoggerDevice/src/YarpRobotLoggerDevice.cpp:413]: We need to consider the comment from @GiulioRomualdi here, https://github.com/ami-iit/bipedal-locomotion-framework/pull/519#issuecomment-1080436810

I was quickly skimming the code of the  [`embObjFTSensor`](https://github.com/robotology/icub-main/tree/26c74355c9d6463762352f41f8f1a1df593844ca/src/libraries/icubmod/embObjFTsensor) yesterday, although I did not come to a complete understanding, I think retrieving the temperature data is done through service calls from the low level. 

cc @traversaro could you shed more light on this, if you know some details about this sensor?

**Comment by prashanthr05** [comment]
> I pushed the modifications discussed yesterday in [af202fb](https://github.com/ami-iit/bipedal-locomotion-framework/commit/af202fb7b9b067f7801eea8b8243f5cea730f1da) and [71d31e2](https://github.com/ami-iit/bipedal-locomotion-framework/commit/71d31e28e5fe8269e217499586f44afc2ef9ac3d).
> 
> cc @prashanthr05

Thanks, @isorrentino for the changes. Looks good. The yarp-telemetry buffer manager in the YarpRobotLogger device needs to be updated too.

File: devices/YarpRobotLoggerDevice/src/YarpRobotLoggerDevice.cpp (hunk 1 above)
Comment (traversaro) [reply devices/YarpRobotLoggerDevice/src/YarpRobotLoggerDevice.cpp:413]: You can call the temperature method anytime you want, as you can see in https://github.com/robotology/icub-main/blob/26c74355c9d6463762352f41f8f1a1df593844ca/src/libraries/icubmod/embObjFTsensor/embObjFTsensor.cpp#L419 the method just access a buffer.

**Comment by isorrentino** [comment]
> Thanks, @isorrentino for the changes. Looks good. The yarp-telemetry buffer manager in the YarpRobotLogger device needs to be updated too.

Fixed in https://github.com/ami-iit/bipedal-locomotion-framework/pull/519/commits/df8e4cf778a70c60263260a93d7586639e7905b1

**Comment by prashanthr05** [comment]
We tested once this PR with changes in [robot-sconfiguration](https://github.com/ami-iit/robots-configuration/commit/8ddc802ab669b4571a33c7ede470ebb1b70d1b97) related to using `embObjFTSensors` instead of `embObjStrain` for the arm FTs on the robot. We were able to access the temperature sensors using he `YarpSensorBridge`.
If we test again with the recent mods introduced by @isorrentino, we may proceed with merging this PR.

cc @HosameldinMohamed

**Comment by isorrentino** [comment]
I have tested the latest changes. Here is the dataset containing the temperature. The configuration files contained a bug that has been fixed in https://github.com/ami-iit/bipedal-locomotion-framework/pull/519/commits/b4e0c8472256473209d15bdbd50ec8772ff31e8d.

[robot_logger_device_2022_05_02_18_24_26.zip](https://github.com/ami-iit/bipedal-locomotion-framework/files/8604289/robot_logger_device_2022_05_02_18_24_26.zip)

cc @prashanthr05 @GiulioRomualdi

**Comment by prashanthr05** [comment]
@GiulioRomualdi @traversaro this PR is ready for review.

**Comment by GiulioRomualdi** [comment]
the visualizer was crashing since thetemperature data is a scalar signal. I fixed it and I released the new version: https://github.com/ami-iit/robot-log-visualizer/releases/tag/v0.1.5

**Pull request merged**